### DISCUSSION TITLE
[Feature / Mod Request] Add Day 2 Rain to Logical & New Areas

### DISCUSSION THREAD

**Discussion by SuperBro64** [HarbourMasters/2ship2harkinian#566] [Ideas]
I fully understand that this would be a very minor change that not many people would care about, but I still want to suggest it in hopes that it could happen in some way.

**Explanation**
In the vanilla game, on Day 2 from 7 AM to 5 PM, it rains naturally in the following areas:
- Clock Town (all sections)
- Southern Swamp (both poisoned and cleansed states)
- Woods of Mystery
- Deku Palace (both poisoned and cleansed states)
- Romani Ranch
- Road to Southern Swamp (only near the entrance to Southern Swamp)
- Milk Road (only near the entrance to Romani Ranch)

However, it does _not_ rain in Termina Field nor Woodfall during this time. This is a discrepancy that has always lowkey bugged me about the original game. Clock Town is in the very center of Termina Field, and Woodfall is in the very center of Southern Swamp. Why is the sky raining over _only_ the specific area where Clock Town is and not anywhere else around it, and vice versa for Woodfall? Similarly, why is it not raining in Gorman Track despite it being near Romani Ranch?

Additionally, I think it would be more interesting and still logical for it to also rain on Day 2 in all of the exterior areas in the Great Bay region, which is near the Southern Swamp and Romani Ranch regions on the map.

**Request**
Thus, what I would want for this request would be an optional toggle, or a mod if it needs to be one, that would make it rain naturally everywhere in the scene on Day 2 from 7 AM to 5 PM in the following areas:
- Termina Field
- Road to Southern Swamp (everywhere)
- Woodfall (both poisoned and cleansed states)
- Milk Road (everywhere)
- Gorman Track
- Great Bay Coast (both misty and cleansed states)
- Zora Cape (both misty and cleansed states)
- Waterfall Rapids
- Pirates' Fortress (entrance section)
- Pirates' Fortress (exterior section)
- Pinnacle Rock

With this change, there would be more of a logical consistency in which areas have rain on Day 2. Again, I know this is minor, but it's something that I honestly would appreciate quite a lot.

**Comment by Dumbledork01** [comment]
I'm guessing any fixes for this would be in this area of the code: https://github.com/HarbourMasters/2ship2harkinian/blob/e484e0ba825e67402e4ca0e2b49eaaa2c2fbecdc/mm/src/overlays/actors/ovl_En_Weather_Tag/z_en_weather_tag.c#L452C1-L452C20

**Reply by SuperBro64** [reply]
I checked it out, and it looks like that snippet of code you linked to concerns what I mentioned about Road to Southern Swamp and Milk Road, where it does not rain everywhere in the scenes but instead starts raining when you approach Southern Swamp or Romani Ranch.

The change I was suggesting was more to make the areas I mentioned have rain across the whole scene like how Clock Town, Southern Swamp, and Romani Ranch do, rather than in only a specific spot of the scene.

Termina Field, Woodfall (poisoned and cleaned), and Gorman Track would get rain everywhere, which means by extension for logical reasons Road to Southern Swamp and Milk Road would also get rain everywhere. I think this would thus require changes to these scenes themselves, compared to the local weather changes which I think is just caused by an invisible object (could be wrong). 

This is why this change might require a mod if it can't be a normal toggle. Not completely sure.

**Reply by Dumbledork01** [reply]
Damn, fair enough. Yeah, Idk where the line between enhancement and mod is haha.

**Comment by SuperBro64** [comment]
I figured out how to mod this request into 2Ship myself as I explained here:
(https://discord.com/channels/808039310850130000/1244457959484821596/1250218751585026109)

So I've decided to close this as resolved.

**Comment by smashbro596** [comment]
to add to your list:
mountain village (spring)
road to goron village (spring)
goron race track (spring)
goron village (spring)

day 2 rain would only occur in the snowhead region after goht is defeated on the cycle.

**Reply by SuperBro64** [reply]
I actually did initially think about having Day 2 rain in every outdoor scene of the game. But then I decided that might have been a bit overkill.

I feel just having the rain in the central, Romani Ranch, Southern Swamp, and Great Bay regions is enough, and leave the Snowhead and Ikana regions alone.
